src/clr-angular/data/datagrid/providers/columns.service.ts

```diff
@@ -30,6 +30,9 @@ export class ColumnsService implements OnDestroy {
 
   // Helper method to emit a change to a column only when there is an actual diff to process for that column
   emitStateChange(columnIndex: number, diff: Partial<DatagridColumnState>) {
+    if (!this.columns[columnIndex]) {
+      this.columns[columnIndex] = new BehaviorSubject<DatagridColumnState>({});
```

Comment:
I'm trying to understand the point of this. I'm guessing there might be a race condition somewhere where we try to change the state of a column before the datagrid's main renderer has the actual list of all the columns.

But shouldn't we just return in this case? We're creating a BehaviorSubject that no one subscribes to, since the `setupColumns()` method of the main renderer just replaces this subject with a new one.

Reply:
Yeah we can just `return`. Fixed.